bundles/org.openhab.binding.pjlinkdevice/src/main/java/org/openhab/binding/pjlinkdevice/internal/PJLinkDeviceHandler.java

```diff
@@ -0,0 +1,329 @@
+                    : OnOffType.OFF);
+          } else {
+            logger.debug("Received unknown power command" + command);
+          }
+          break;
+        case CHANNEL_INPUT:
+          if (command == RefreshType.REFRESH) {
+            StringType input = new StringType(device.getInputStatus().getResult().getValue());
+            updateState(PJLinkDeviceBindingConstants.CHANNEL_INPUT, input);
+          } else if (command instanceof StringType) {
+            logger.trace("Received input command" + command);
+            Input input = new Input(((StringType) command).toString());
+            device.setInput(input);
+          } else {
+            logger.debug("Received unknown channel command" + command);
+          }
+          break;
+        case CHANNEL_AUDIO_MUTE:
+        case CHANNEL_VIDEO_MUTE:
+          boolean isAudioMute = channelUID.getId().equals(PJLinkDeviceBindingConstants.CHANNEL_AUDIO_MUTE);
+          boolean isVideoMute = channelUID.getId().equals(PJLinkDeviceBindingConstants.CHANNEL_VIDEO_MUTE);
+          if (isVideoMute || isAudioMute) {
+            if (command == RefreshType.REFRESH) {
+              // refresh both video and audio mute, as it's one request
+              MuteQueryResponseValue muteStatus = device.getMuteStatus();
+              updateState(PJLinkDeviceBindingConstants.CHANNEL_AUDIO_MUTE,
+                  muteStatus.isAudioMuted() ? OnOffType.ON : OnOffType.OFF);
+              updateState(PJLinkDeviceBindingConstants.CHANNEL_VIDEO_MUTE,
+                  muteStatus.isVideoMuted() ? OnOffType.ON : OnOffType.OFF);
+            } else {
+              if (isAudioMute) {
+                logger.trace("Received audio mute command" + command);
+                boolean muteOn = command == OnOffType.ON;
+                device.setMute(MuteInstructionChannel.AUDIO, muteOn);
+              }
+              if (isVideoMute) {
+                logger.trace("Received video mute command" + command);
+                boolean muteOn = command == OnOffType.ON;
+                device.setMute(MuteInstructionChannel.VIDEO, muteOn);
+              }
+            }
+          }
+      }
+
+      updateStatus(ThingStatus.ONLINE);
+      logger.trace("Successfully handled command {} on channel {}", command, channelUID.getId());
+    } catch (IOException e) {
+      updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.COMMUNICATION_ERROR, e.getMessage());
+    } catch (ResponseException e) {
+      updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.COMMUNICATION_ERROR, e.getMessage());
+    } catch (ConfigurationException e) {
+      updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.CONFIGURATION_ERROR, e.getMessage());
+    } catch (AuthenticationException e) {
+      handleAuthenticationException(e);
+    }
+  }
+
+  @Override
+  public void initialize() {
+    this.setupJob = scheduler.schedule(() -> {
```

Comment:
```suggestion
    this.setupJob = scheduler.execute(() -> {
```

Reply:
The issue with execute() is that it does not return a ScheduledFuture, which means I cannot stop the created job in dispose()/clearSetupJob()

Reply:
Why would you want to cancel a one-time job?

Reply:
@nils Can you comment on this one? There are only two points open (the other being the configuration in initialize instead of later method calls) before we can merge. We might get that into M3 then.

Reply:
In case dispose() is called immediately after initialize().

Reply:
I doubt this will work. `schedule` with a zero delay will start the task immediately. Calling `cancel` prevents future executions, but does not guarantee that a running task stops. We use `execute` in other places, too. But I would not make my approval dependent on this change.

Reply:
The need for being able to stop the scheduled connection attempt just started to make sense due to the new autoReconnectPeriod the delay is now also used for delayed reconnections (it's only 0 on first attempt to connect).